PhysicsTools/PatExamples/plugins/MuonVIDCITKAnalyzer.cc

```diff
@@ -0,0 +1,206 @@
+
+#include "TH1F.h"
+#include "TH2F.h"
+
+#include <sstream>
+
+using namespace std;
+using namespace edm;
+//
+// class decleration
+//
+class MuonVIDCITKAnalyzer : public edm::stream::EDAnalyzer<>
+{
+public:
+  MuonVIDCITKAnalyzer(const edm::ParameterSet&);
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+private:
+  edm::EDGetTokenT<edm::View<reco::Candidate> > muonToken_;
+  edm::EDGetTokenT<reco::VertexCollection> vertexToken_;
+
+  // VID
+  typedef edm::ValueMap<bool> VIDMap;
+  edm::EDGetTokenT<VIDMap> muonLooseVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonMediumVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonTightVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonSoftVIDToken_;
+  edm::EDGetTokenT<VIDMap> muonHighPtVIDToken_;
+
+  // CITK
+  typedef edm::ValueMap<float> CITKMap;
+  edm::EDGetTokenT<CITKMap> muonChIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonNhIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonPhIsoCITKToken_;
+  edm::EDGetTokenT<CITKMap> muonPuIsoCITKToken_;
+
+  // IsoDeposit
+  typedef edm::ValueMap<double> IsoMap;
+  edm::EDGetTokenT<IsoMap> muonChIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonNhIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonPhIsoIsoDepToken_;
+  edm::EDGetTokenT<IsoMap> muonPuIsoIsoDepToken_;
+
+  // Histograms
+  TH2F* h2Ch_, * h2Nh_, * h2Ph_, * h2Pu_;
+  TH1F* hIsoDiffCh_, * hIsoDiffNh_, * hIsoDiffPh_, * hIsoDiffPu_;
+};
+
+MuonVIDCITKAnalyzer::MuonVIDCITKAnalyzer(const edm::ParameterSet& iConfig)
+{
+  muonToken_ = consumes<edm::View<reco::Candidate> >(edm::InputTag("muons"));
+  vertexToken_ = consumes<reco::VertexCollection>(edm::InputTag("offlinePrimaryVertices"));
+
+  string vidPrefix = "muoMuonIDs:cutBasedMuonId-MuonPOG-V0-";
+  muonLooseVIDToken_  = consumes<VIDMap>(edm::InputTag(vidPrefix+"loose" ));
+  muonMediumVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"medium"));
+  muonTightVIDToken_  = consumes<VIDMap>(edm::InputTag(vidPrefix+"tight" ));
+  muonSoftVIDToken_   = consumes<VIDMap>(edm::InputTag(vidPrefix+"soft"  ));
+  muonHighPtVIDToken_ = consumes<VIDMap>(edm::InputTag(vidPrefix+"highpt"));
+
+  muonChIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h+-DR040-ThresholdVeto000-ConeVeto000"));
+  muonNhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:h0-DR040-ThresholdVeto050-ConeVeto000"));
+  muonPhIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFNoPileUpIsolation:gamma-DR040-ThresholdVeto050-ConeVeto000"));
+  muonPuIsoCITKToken_ = consumes<CITKMap>(edm::InputTag("muonPFPileUpIsolation:h+-DR040-ThresholdVeto050-ConeVeto000"));
+
+  muonChIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueCharged04PAT"));
+  muonNhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueNeutral04PAT"));
+  muonPhIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValueGamma04PAT"));
+  muonPuIsoIsoDepToken_ = consumes<IsoMap>(edm::InputTag("muPFIsoValuePU04PAT"));
+
+  edm::Service<TFileService> fs;
```

Comment:
This should be changed to edm::one::EDAnalyzer.
Based on implementation (Fill is called every event, nothing special about run or lumi block transitions) use
``` edm::one::EDAnalyzer<edm::one::SharedResources> ```